### PR TITLE
Dts stn/dums 75 subscription mutations

### PR DESCRIPTION
**Pull request by alexunique0519**
## [DUMS-XXX](https://jira-dev.bdm-dev.dts-stn.com/browse/DUMS-XXX) (Jira Issue)
[DUMS-75](https://jira-dev.bdm-dev.dts-stn.com/browse/DUMS-75)
### Description
Added showHideRoomCard subscription such that once cards' show/hide value changes, it will notify all clients 
List of proposed changes:

-
-

### CODE REVIEW

File: api/room/mutations.js [hunk 1]
```diff
@@ -39,4 +42,16 @@ export default {
       return { success: false, message: "Failed to add user" };
     }
   },
+  showHideRoomCard: async (_, { roomId, isShown }) => {
```
Comment (DaceyTom2) [inline api/room/mutations.js]: I'd recommend using the updateRoom mutator instead of creating one specific to show/hide room

**Review by DaceyTom2 — CHANGES_REQUESTED** [review]
Looks really good, the only thing would be changing the subscription and mutation to follow an updateRoom pattern rather than being specific to show/hide

File: api/room/mutations.js (hunk 1 above)
Comment (alexunique0519) [reply api/room/mutations.js]: Changed to use updateRoom mutation

File: api/room/mutations.js [hunk 2]
```diff
@@ -18,12 +21,21 @@ export default {
...
+        const room = getRoomByID(id)
+        pubsub.publish("ROOM_UPDATED", {
+          roomUpdated: room
+        });
+        return { success: true, message: "Room updated" };
+      } else {
+        return { success: false, message: "Failed to update room" };
+      }
+    } catch (error) {
+      console.log('update room error', error);
```
Comment (DaceyTom2) [inline api/room/mutations.js]: This can return
return { success: false, message: "Failed to update room" };

instead of logging

**Review by DaceyTom2 — CHANGES_REQUESTED** [review]
Looking good! Update the console log and ++

**Review by DaceyTom2 — APPROVED** [review]
Great work!

**Pull request merged**